pkg/scheduler/apis/config/v1alpha2/defaults.go

```diff
@@ -36,9 +37,15 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 
 // SetDefaults_KubeSchedulerConfiguration sets additional defaults
 func SetDefaults_KubeSchedulerConfiguration(obj *v1alpha2.KubeSchedulerConfiguration) {
-	if obj.SchedulerName == nil {
-		val := api.DefaultSchedulerName
-		obj.SchedulerName = &val
+	if len(obj.Profiles) == 0 {
+		obj.Profiles = append(obj.Profiles, v1alpha2.KubeSchedulerProfile{})
+	}
+	for i := range obj.Profiles {
+		profile := &obj.Profiles[i]
+		if profile.SchedulerName == nil {
+			// Validation ensures that only one profile uses DefaultSchedulerName.
+			profile.SchedulerName = pointer.StringPtr(api.DefaultSchedulerName)
```

Comment:
would it make more sense to only apply a default name if there's a single profile, and leave it undefaulted (and make validation fail with a "required" error) if there's more than a single profile?

Reply:
Yes, this gives a more comprehensible error :)
Done